lib/json-schema/attributes/formats/ip.rb

```diff
@@ -11,7 +11,7 @@ def self.validate(current_schema, data, fragments, processor, validator, options
         begin
           ip = IPAddr.new(data)
         rescue ArgumentError => e
-          raise e unless e.message == 'invalid address'
+          raise e unless e.message.start_with?('invalid address')
         end
```

Comment:
I feel that parsing exceptions is fragile and I'd implement it like this (roughly):
```ruby
class IPFormat < FormatAttribute
  def self.validate(current_schema, data, fragments, processor, validator, options = {})
    return unless data.is_a?(String)

    unless self.regex.match?(data)
      error_message = ....
      validation_error(...)
    end
  end

  def self.regex
    raise NotImplementedError
  end
end

class IP4Format < IPFormat
  def self.ip_version
    4
  end

  def self.regex
    Resolv::IPv4::Regex
  end
end
```

I do think `regex.match?()` may be Ruby 2.5 only but you can use `=~` for older Rubies.

However, I'm not sure if this regex is available in older Rubies.